server/data/comments.js

```diff
@@ -1,9 +1,47 @@
 const { ObjectId } = require('mongodb');
 const { getProjectById } = require('./projects');
 const { projects } = require('../config/mongoCollections');
-const { isValidStr, internalServerErr, isValidObjectId } = require('../utils');
+const {
+	isValidStr,
+	internalServerErr,
+	isValidObjectId,
+	notFoundErr,
+	forbiddenErr,
+} = require('../utils');
 const { isValidUsername } = require('../utils/users');
 
+const getCommentById = async (commentId) => {
+	const id = isValidObjectId(commentId);
+	const projectsCollection = await projects();
+	const comment = await projectsCollection.findOne({
+		comments: { $elemMatch: { _id: ObjectId(id) } },
+	});
+	if (!comment) throw notFoundErr('No comment found for the provided id');
+	return comment;
+};
+
+const iscommentOwner = async (projectId, commentId, userId) => {
+	let isOwner = false;
+	projectId = isValidObjectId(projectId);
+	commentId = isValidObjectId(commentId);
+	userId = isValidObjectId(userId.toString());
+	const projectFind = await getProjectById(projectId);
+	if (projectFind.owner._id.toString() === userId) {
+		isOwner = true;
+	}
+	let commentsArray = projectFind.comments;
+	for (let comment in commentsArray) {
```

Comment:
i think using forEach would be better for performance. did you not get an ESLint error for this? 
or you could use array.find function to find a comment with the id and then compare the owner id

Reply:
The problem is since I used `prettier/prettier: ['error', { 'endOfLine': 'auto' }]` in .yml file, it ignored all the errors. I commented it and checked for all the errors now. Errors are fixed.

I used forEach now. This is done.